src/generate_list.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+################################################################################
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2017 Massachusetts Institute of Technology.
+################################################################################
+
+# Configure the installer here
+INITRAMFS_TOOLS_GIT=https://salsa.debian.org/kernel-team/initramfs-tools.git
+INITRAMFS_TOOLS_VER="master"
+
+
+# Grabs Debian's initramfs_tools from Git repo if no other options exist
+if [[ ! `command -v unmkinitramfs` && ! -x "/usr/lib/dracut/skipcpio" ]] ; then
+    # Create temp dir for pulling in initramfs-tools
+    TMPDIR=`mktemp -d` || exit 1
+    echo "INFO: Downloading initramfs-tools: $TMPDIR"
+
+    # Clone initramfs-tools repo
+    pushd $TMPDIR
+    git clone $INITRAMFS_TOOLS_GIT initramfs-tools
+    pushd initramfs-tools
+    git checkout $INITRAMFS_TOOLS_VER
+    popd # $TMPDIR
+    popd
+
+    shopt -s expand_aliases
+    alias unmkinitramfs=$TMPDIR/initramfs-tools/unmkinitramfs
+fi
+
+
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root" 1>&2
+    exit 1
+fi
+
+if [ $# -lt 1 ]
+then
+    echo "Usage:  `basename $0` list.txt [hash-algo]" >&2
+    exit $NOARGS;
+fi
+
+# Where to look for initramfs image
+INITRAMFS_LOC="/boot/"
+if [ -d "/ostree" ]; then
+    # If we are on an ostree system change where we look for initramfs image
+    loc=$(grep -E "/ostree/[^/]([^/]*)" -o /proc/cmdline | head -n 1 | cut -d / -f 3)
+    INITRAMFS_LOC="/boot/ostree/${loc}/"
+fi
+
+if [ $# -eq 2 ]
+then
+    ALGO=$2
+else
+    ALGO=sha1sum
+fi
+
+OUTPUT=$(readlink -f $1)
+rm -f $OUTPUT
+
+
+echo "Writing whitelist to $OUTPUT with $ALGO..."
+
+# Add all appropriate files under root FS to whitelist
+cd /
+find `ls / | grep -v "\bsys\b\|\brun\b\|\bproc\b\|\blost+found\b\|\bdev\b\|\bmedia\b\|\bsnap\b\|mnt"` \( -fstype rootfs -o -xtype f -type l -o -type f \) -uid 0 -exec $ALGO '/{}' >> $OUTPUT \;
+
+# Create staging area for init ram images
+rm -rf /tmp/ima/
```

Comment:
I'd try hard to avoid predictable filenames in `/tmp`: https://cwe.mitre.org/data/definitions/377.html
A common pattern I use is:
`cd $(mktemp -d)`
(though this misses cleaning up the dir afterwards; to fix that it's usually simplest to use systemd and `PrivateTmp=true` or so in the unit)